### PR TITLE
Improve feature gate StatefulSetAutoDeletePVC description

### PR DESCRIPTION
**Pull request by mengjiao-liu**
Improve feature gate StatefulSetAutoDeletePVC description.

Ref: https://github.com/kubernetes/kubernetes/issues/113319#issuecomment-1297946435
> It is best to have a place for the document to be queried by the administrator and avoid confusion.

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 1]
```diff
@@ -350,8 +350,8 @@ StatefulSet will then begin to recreate the Pods using the reverted template.
 The optional `.spec.persistentVolumeClaimRetentionPolicy` field controls if
 and how PVCs are deleted during the lifecycle of a StatefulSet. You must enable the
 `StatefulSetAutoDeletePVC` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
-to use this field. Once enabled, there are two policies you can configure for each
-StatefulSet:
+on the API server and the controller manager to use this field.
```
Comment (mattcary) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:353]: Would it be safer to say to enable the feature gate on all k8s components? And in general isn't it best practice to have feature gates consistent across all components anyway? (This last question directed more at website approvers who might know standard practice better).

Comment (mattcary) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:353]: As @mengjiao-liu pointed out, other features like https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-tracking-with-finalizers specify the components to set the feature gate on. I'm not sure I agree with that, but I'll defer to standard practice.

/lgtm

Comment (mengjiao-liu) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:353]: Seek more advice
 /cc @sftim  @tengqm

**Comment by tengqm** [comment]
/approve

Features can be enabled on a per-component basis, though it adds burden for operation. Clarifying the components doesn't hurt though, and for some use cases, it might be useful.

**Pull request merged**